src/ORM/RulesChecker.php

```diff
@@ -71,14 +71,25 @@ public function isUnique(array $fields, $message = null)
      * $rules->add($rules->existsIn('site_id', new SitesTable(), 'Invalid Site'));
      * ```
      *
+     * Available $options are error 'message' and 'allowSqlNulls' flag.
+     *
      * @param string|array $field The field or list of fields to check for existence by
      * primary key lookup in the other table.
      * @param object|string $table The table name where the fields existence will be checked.
-     * @param string|null $message The error message to show in case the rule does not pass.
+     * @param array|string|null $options List of options or error message string to show in case the rule does not pass.
      * @return callable
      */
-    public function existsIn($field, $table, $message = null)
+    public function existsIn($field, $table, $options = null)
     {
+        $message = null;
+        if (is_string($options)) {
+            $message = $options;
+            $options = null;
+        } elseif (is_array($options)) {
+            $options += ['message' => ''];
+            $message = $options['message'];
+            unset($options['message']);
+        }
```

Comment:
Following would be much more readable:

```php
if (is_string($options)) {
    $options = ['message' => $options];
}

$options = (array)$options + ['message' => null];
$message = $options['message'];
unset($options['message']);
```

Would also remove the need to typecast $options to array on line 102.